pkg/controllers/infrastatus/reconcile.go

```diff
@@ -0,0 +1,78 @@
+package infrastatus
+
+import (
+	"fmt"
+	"reflect"
+	"sync"
+
+	"github.com/go-logr/logr"
+
+	kubeclient "k8s.io/client-go/kubernetes"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	configv1 "github.com/openshift/api/config/v1"
+
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	configlister "github.com/openshift/client-go/config/listers/config/v1"
+)
+
+type InfraStatusReconciler struct {
+	Source     *configv1.Infrastructure
+	Client     configclient.Interface
+	KubeClient kubeclient.Interface
+	Lister     configlister.InfrastructureLister
+	Log        logr.Logger
+
+	m              sync.Mutex
+	hasSubresource *bool
+}
+
+func (r *InfraStatusReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	proceed, err := r.shouldReconcile()
+	if err != nil || !proceed {
+		return ctrl.Result{}, err
+	}
+	if req.Name != "cluster" {
```

Comment:
If this condition is true, it would be unusual, right? Should we log something here?